### DISCUSSION TITLE
'Force Run' RSS Feed via API?

### DISCUSSION THREAD

**Discussion by marksink** [autobrr/autobrr#1555] [Q&A] [answered]
The UI allows you to select an RSS feed, and issue a 'Force Run' to trigger it to update.  Has anyone figured out how to accomplish this via the API?

**Comment by KyleSanderson** [comment]
What are you trying to do?

**Reply by marksink** [reply]
'Force Run' RSS Feed via API

**Reply by s0up4200** [reply, marked as answer]
To force run (remember this happens on a set interval already, so don't overdo it), simply:
```shell
curl -X POST 'http://127.0.0.1:7474/api/feeds/ID/forcerun' -H 'X-API-Token: AUTOBRR_API_KEY'
```
Where ID is the number of the feed you wish to force run.

Feed IDs can be fetched with:
```shell
curl -X GET 'http://127.0.0.1:7474/api/feeds' -H 'X-API-Token: AUTOBRR_API_KEY' | jq '.[] | {id, name, enabled}'
```

Don't get yourself banned

**Reply by marksink** [reply]
I had tried /force_run  (underscore included) just guessing what it might be, but even /forcerun doesn't show by looking at 'View latest run' that it is actually being run. Last run in UI, or output from 'View Last Run' is still old. So not sure if this is working. Im getting the ID from the View Latest Run ID shown in the URL

**Reply by s0up4200** [reply]
UI might not update when running it like that. The logs should confirm it being run though.

**Reply by KyleSanderson** [reply]
What are you trying to do.

**Comment by n1ce-0ne** [comment]
sometimes i get HTTP ERROR 405 not sure why

**Reply by zze0s** [reply]
Can you try and replicate under what circumstances it happens?

**Reply by n1ce-0ne** [reply]
thankyou for the reply,

it seems to happen once every roughly each month.
im not sure whats causing it.
i can use both my instances to check between to see if each act differently when one seems to act up. [local/seedbox]
since i posted that above ive realised when running /api/feeds/1/forcerun or /api/feeds/1/cache in the browser i always get HTTP ERROR 405 even when its working so im not sure if that intentional or not when running the query in the browser.

i basically use php for using the api with curl commands using the $headers = array("X-API-Token: <myApiKey>");
i use the api to send torrent files to rutorrent/rtorrent, 1 at a time with the click of a js button on the php page.
on each click of the button it creates an .xml file aka an rss file which contains all the details needed for autobrr to then match a filter & send it to rtorrent, once the .xml file is ready i then use the api to 'forcerun' the feed then clear the 'cache' with the 2x commands /api/feeds/1/forcerun & /api/feeds/1/cache, which i have to say works brilliantly until the bug im mentioning.

yesterday while the bug was going on i restarted autobrr which didn't make a difference, i also tried duplicating everything so i can test with /api/feeds/2/forcerun & /api/feeds/2/cache but the issue still persisted on both 1 & 2 when using php curl, so tried with /api/feeds to see if that worked or not, because i use $response = curl_exec($ch); echo $response while debugging , but /api/feeds didnt work either so it seems its nothing related to specific indexer entries but rather the whole api, i.e curl just simply cannot communicate/reach the api, i tried CURLOPT_VERBOSE but that didnt help, i tried CURLINFO_HEADER_OUT with var_dump(curl_getinfo($ch,CURLINFO_HEADER_OUT)); to see if that could reveille anything, that was basically 'bool(0); or similar when it wouldn't connect but then testing with my local instance it returned my api key etc so it just confirmed it wasn't making the connection for some reason.

as a temp work-a-round i just set the rss refresh interval to 1 minute which works but of course if a new .xml file is created more than once in that minute only the last file is send to rtorrent, and of course grabbing the xml file every minute isn't too good,

i was checking the logs but i forgot to set to trace so it was looking like nothing was happening but next time it happens i'll deffo set to trace and see if i can get more info 👀

**Reply by zze0s** [reply]
405 is usually `method not allowed`. Do you trigger it with a POST?

**Reply by n1ce-0ne** [reply]
ahh so thats why we get 405 in the browser, yes with php curl i have: 
if (strpos($action,'/forcerun') !== false) curl_setopt($ch,CURLOPT_POST, 1);  //0 for a get request
elseif (strpos($action,'/cache') !== false) curl_setopt($ch,CURLOPT_CUSTOMREQUEST, "DELETE");
else curl_setopt($ch,CURLOPT_POST, 0);

it shouldn't be the php code because it works perfectly for a good while then approx once a month for like less than 24h it just totally stops working.

all i can think is next time it happens ill enable the trace logs see if they show up anything & ill try the api call with bash curl with -v flag because i may get a better output from the curl call with some better idea on why the curl call isn't working.
